app/models/work.rb

```diff
@@ -46,17 +46,38 @@ def works_by_user_state(user, state)
       works = []
       if user.admin_collections.count == 0
         # Just the user's own works by state
-        works = Work.where(created_by_user_id: user, state: state)
+        works += Work.where(created_by_user_id: user, state: state)
       else
         # The works that match the given state, in all the collections the user can admin
         # (regardless of who created those works)
         user.admin_collections.each do |collection|
           works += Work.where(collection_id: collection.id, state: state)
         end
       end
+
+      # Any other works where the user is mentioned
+      works_mentioned_by_user_state(user, state).each do |work_id|
+        already_included = !works.find { |work| work[:id] == work_id }.nil?
+        works << Work.find(work_id) unless already_included
+      end
+
       works.sort_by(&:updated_at).reverse
     end
 
+    # Returns an array of work ids where a particular user has been mentioned
+    # and the work is in a given state.
+    def works_mentioned_by_user_state(user, state)
+      sql = <<-END_SQL
+        SELECT DISTINCT works.id
+        FROM works
+        INNER JOIN work_activities ON works.id = work_activities.work_id
+        INNER JOIN work_activity_notifications ON work_activities.id = work_activity_notifications.work_activity_id
+        WHERE work_activity_notifications.user_id = #{user.id} AND works.state = '#{state}'
+      END_SQL
+      rows = ActiveRecord::Base.connection.execute(sql)
+      rows.map { |row| row["id"] }
+    end
+
```

Comment:
See our ERD if this SQL is puzzling: https://docs.google.com/drawings/d/1q2sfj8rrcNVgqQPK5uT_t79A9SYqncinh3HbnCSGMyQ/edit